src/python/pants/core/goals/tailor.py

```diff
@@ -285,6 +285,36 @@ def register_options(cls, register):
             f"to the one it replaces (see {doc_url('macros')}).",
         )
 
+        register(
+            "--ignore-paths",
+            advanced=True,
+            type=list,
+            member_type=str,
+            help=(
+                "Do not edit or create BUILD files at these paths.\n\n"
+                "Can use literal file names and/or globs, e.g. "
+                "`['project/BUILD, 'ignore_me/**']`.\n\n"
+                "This augments the option `[GLOBAL].build_ignore`, which tells Pants to also not "
+                "_read_ BUILD files at certain paths. In contrast, this option only tells Pants to "
+                "not edit/create BUILD files at the specified paths."
+            ),
+        )
+        register(
+            "--ignore-adding-targets",
+            advanced=True,
+            type=list,
+            member_type=str,
+            help=(
+                "Do not add these target definitions.\n\n"
+                "Expects a list of target addresses that would normally be added by `tailor`, "
+                "e.g. [`project:tgt']`. To find these names, you can run `tailor`, then combine "
+                "the BUILD file path with the target's name. For example, if `tailor` is adding "
+                "the target `bin` to `project/BUILD`, then the address would be `project:bin`. If "
+                "the BUILD file is at the root of your repository, use `//` for the path, e.g. "
+                "`//:bin`."
```

Comment:
This is definitely a bit wonky, but I think it's less offensive than changing our update for `tailor` to start saying the full address path. That gets really really verbose, I tried it :)

Reply:
Should this support regex matches or some sort of blob matching?

What about filtering out specific target types?

Reply:
> Should this support regex matches or some sort o glob matching?

Maybe, but I suspect at that point you'd use path ignores? I'd rather bias towards simplicity now. If we decide to do regex ignores, we could do what `--ignore-warnings` does and require `$regex$` as a prefix.

> What about filtering out specific target types?

I think that should probably be a dedicated option? Can add if people are asking for it in followups.